### ISSUE TITLE
Crazy idea for 2020

### ISSUE THREAD

**Issue by unknowndomain** [emfcamp/Website#795]
The problems at the bar this year with card payments were frustrating, but it highlighted that the entire card payment process is frustrating, one person suggested writing an integration so that the card readers get sent the payment amount, and capture the authorisation codes, but another idea was floated...

Could 2020's website use Stripe or similar to hold the customers card details for recurring payments, and then we build some kind of integration where the wrist bands are RFID tags and you pay using your wrist band during the event.

Would this mean cash back would not be possible, can an online card payment be completed in a way that ensures we're getting cash? Would the processing fees be too high?

Basically the idea is that the card details are entered into Stripe so we never handle them directly and that the wrist band is associated with the ticket when you get to the gate.

**Comment by jellybob** [comment]
Cash back would certainly be possible, I think the danger we'd have would be ensuring attendees actually have the money needed to pay their bill at the end of the weekend (assuming we effectively let everyone run a tab for the duration of the event).

My inclination would be to run more of a pre-pay system, where attendees can choose to add an amount of money to their account via Stripe or even by handing over cash at the bar, and then spend that across the course of the weekend - we could even give people the option at the end of the event of either getting a refund on their remaining balance, or donating that towards the running of future events, which I suspect quite a few people would be happy to do.

In principle though I love the idea, and I think it would be super popular with the sort of people who show up.

**Comment by jellybob** [comment]
In summary please take that as a "hello, I'm volunteering to build that, because it sounds fun".

**Comment by Pezmc** [comment]
You've just invented [Festipay](http://www.festipay.com/)! Top up before you arrive/at a cash desk, spend money using RFID + QR, cash out when you leave.

That's not to say I don't think it would be handy!

**Comment by sde1000** [comment]
> one person suggested writing an integration so that the card readers get sent the payment amount, and capture the authorisation codes

This is on my TODO list; the problem is finding card reader manufacturers / suppliers who are willing to work with you.  So far the only support I've had is from Verifone, and using their terminals also means subscribing to their "WebCom" service which routes payment requests from their terminals to acquiring banks.  Verifone terminals are also quite pricey!

**Comment by russss** [comment]
I don't think prepay would work because this would be classed as a "cash advance" transaction (legally subject to fees on credit cards) which Stripe doesn't support to my knowledge. 

Cashback (which is effectively a partial cash advance transaction) likewise needs labelling and I don't believe Stripe supports that either.

Additionally I don't think we want to require all attendees to carry RFID fobs or similar.

So you're stuck with authorising payments through Stripe in realtime - which is *probably* permitted - but in any case this means we're managing a lot of payment infrastructure.

**Comment by russss** [comment]
And Steve has just covered the reason why we don't have integration between the card terminals and quicktill.

The other option is that we drop quicktill and rent a commercially-available POS system, but I'm disinclined to pursue this option.

**Comment by sde1000** [comment]
I reckon there's a 50% chance I'll have the Verifone integration done and signed off by Verifone by next EMF.  We could look at renting some VX680s.  Apart from the convenience of integration, the other bonus is that you don't have to print and store merchant copies of receipts because they are stored online by Verifone.  (I don't know how much they charge for this on "dormant" accounts; on live accounts it's ~£12/month/terminal.)

**Comment by unknowndomain** [comment]
Having a card reader on loan for a longer period, or even owning one might be worth considering if we are going to need to do mini bars for EMP events. It would then allow for development work between events.

**Comment by marksteward** [comment]
I own a card reader, but AIUI it's the apps that determine how to integrate with it, so depends heavily on who we choose to provide the service.

**Comment by sde1000** [comment]
Officially the protocol you need to use to communicate with a card terminal is dependent on the vendor and on the software installed on it. Unofficially I'm aware that there is a *lot* of overlap between them. You still need to get your point of sale system certified by each manufacturer and bank that you want to use.

I have a developer mode VX680 at home to work on for quicktill.

**Comment by russss** [comment]
I'm closing this due to the reasons stated above.

At the very least, for 2020 we'll ask for at least some ethernet/bluetooth card machines for the bar which will hopefully not suffer from the wifi issues we saw in 2018.
